core/trino-main/src/test/java/io/trino/sql/gen/BenchmarkDynamicPageFilter.java

```diff
@@ -0,0 +1,246 @@
+import io.trino.operator.project.SelectedPositions;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.LazyBlock;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.TestingColumnHandle;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.predicate.ValueSet;
+import io.trino.spi.security.ConnectorIdentity;
+import io.trino.spi.type.Type;
+import io.trino.sql.gen.columnar.FilterEvaluator;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.jmh.Benchmarks.benchmark;
+import static io.trino.operator.project.SelectedPositions.positionsRange;
+import static io.trino.spi.predicate.Domain.DiscreteSet;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TypeUtils.readNativeValue;
+import static io.trino.spi.type.TypeUtils.writeNativeValue;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.util.DynamicFiltersTestUtil.createDynamicFilterEvaluator;
+import static java.util.Objects.requireNonNull;
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+@State(Thread)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Fork(2)
+@Warmup(iterations = 10, time = 1)
+@Measurement(iterations = 15, time = 1)
+public class BenchmarkDynamicPageFilter
+{
+    private static final int MAX_ROWS = 200_000;
+    private static final FullConnectorSession FULL_CONNECTOR_SESSION = new FullConnectorSession(
+            testSessionBuilder().build(),
+            ConnectorIdentity.ofUser("test"));
+
+    @Param(".05")
+    public double inputNullChance = 0.05;
+
+    @Param("0.2")
+    public double nonNullsSelectivity = 0.2;
+
+    @Param({"100", "1000", "5000"})
+    public int filterSize = 100;
+
+    @Param("false")
+    public boolean nullsAllowed;
+
+    @Param({
+            "INT32_RANDOM",
+            "INT64_RANDOM",
+            "INT64_FIXED_32K",
+    })
+    public DataSet inputDataSet;
+
+    private List<Page> inputData;
+    private FilterEvaluator filterEvaluator;
+
+    public enum DataSet
+    {
+        INT32_RANDOM(INTEGER, (block, r) -> INTEGER.writeLong(block, r.nextInt())),
+        INT64_RANDOM(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong())),
+        INT64_FIXED_32K(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong() % 32768)), // LongBitSetFilter
```

Comment:
Did you test smaller ranges than 32K? What ranges do we expect in practice?

Reply:
The 32K here is actually just the range of the input values. Narrowing the input range ensures that bitset will be chosen instead of hashset.
Size of set is determined by `filterSize` and for that the benchmark params are 100, 1000, 5000

Reply:
I was asking basically if we expect in practice smaller than 32k value ranges (max - min)?

Reply:
From what i've seen in benchmarks and other anecdotal cases, the joined columns tend to be dates/times or some kind of unique id or primary key which are usually in a not super-wide range.